Week2/QA_PREP_EXERCISE.md

```diff
@@ -0,0 +1,29 @@
+As a preparation step for the upcoming Q&A, you need to work on the following exercise, which is based on the prep exercise of the previous week. 
+
+### Exercise
+
+Last week you designed a database for storing food recipes. How can you improve your tables, based on what you learned this week?
+
+- Where can you add a primary key?
+- Which tables should be linked with a foreign key?
+- What type of relationships do you see between the tables?
+
+- Can you make queries to get:
+    > All the vegetarian recipes with potatoes
+    > All the cakes that do not need baking
+    > All the vegan and Japanese recipes
+
+(Since your table data might be different from the above questions, feel free to improvise.)
```

Comment:
Let's ask them to add relevant data to their tables, so that most students will have similar data and discussions can be more focussed.

Reply:
Think that is a good idea, can we make it a requirement to have the database running locally? And that they need to have these queries in a js file?

Reply:
I have a question here. Are students free to modify the schema at this point?

For example, from this query, it seems like a `requires_baking` field would be necessary to know this, or an alternative is to query and check for steps where `%bake|baking%` is found. I think giving the students the freedom to decide is nice, they will come up with their own solutions and then we guide them into the one that makes sense.

Reply:
I was thinking that this is more like another category, like "no-bake".

Reply:
If we are going to ask them to add relevant data, should we give them links to existing recipes? And should these be added in the first prep-exercise?

Reply:
@GinaSt How about we ask them to modify the DB at this point as per your suggestion https://github.com/HackYourFuture/databases/pull/500#discussion_r830260735. This can be good training as to how it is to "migrate" data in models, going from a single category to multiple.

And yes, we can provide them with a few recipes, and ask them to add additional ones. Would that be sensible?

Reply:
I added the list of recipes in week 2. The idea is that in the first week they create their own data, with a focus on cakes and Japanese recipes and then in week 2 we ask them to add specific recipes which will make them alter their schema a bit. I will try to add some more recipes later today.

@acadavid Is this what you had in mind more or less?